library/ssl_tls.c

```diff
@@ -6978,4 +6978,113 @@ int mbedtls_ssl_get_key_exchange_md_tls1_2( mbedtls_ssl_context *ssl,
 
 #endif /* MBEDTLS_SSL_PROTO_TLS1_2 */
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+
+int mbedtls_ssl_get_handshake_transcript( mbedtls_ssl_context *ssl,
+                                          const mbedtls_md_type_t md,
+                                          unsigned char *dst,
+                                          size_t dst_len,
+                                          size_t *olen )
+{
+    ((void) ssl);
+    ((void) md);
+    ((void) dst);
+    ((void) dst_len);
+    *olen = 0;
+    return( MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE );
+}
+
+#else /* MBEDTLS_USE_PSA_CRYPTO */
+
+typedef union
```

Comment:
As I said in https://github.com/ARMmbed/mbedtls/pull/4927/files#r711995818, this reorganization for Mbed TLS API case does not work well. It makes just things more complicated than they should be. Thus I would suggest eventually to just remove the commit "Refactor get_handshake_transcript" then in a new commit remove completely the PSA specific code for `mbedtls_ssl_get_handshake_transcript()`.